packages/gatsby/src/redux/__tests__/run-sift.js

```diff
@@ -131,6 +131,17 @@ if (!process.env.GATSBY_DB_NODES || process.env.GATSBY_DB_NODES === `redux`) {
         actions.createNode(node, { name: `test` })(store.dispatch)
       )
     })
+    // There was once a bug in filter lte evaluation that only triggered under
```

Comment:
I couldn't trigger the failing case with just the 4 nodes in the original test. Added this machinery to add a few more, which did get me failing tests before the changes. Happy for guidance here.

Reply:
I've added a similar approach in #23637 to the run-query tests. I guess, since the number of nodes were this small, the pivot never really had to be set again and it didn't lead to a problem. The set was just too small. So I've added some range checks to a set of 100 nodes as well (and checked that they caught this infinite loop as well before fixing it). Thanks for the inspiration.

I feel we don't have to add it here anymore. What do you think?